packages/core-components/src/components/DismissableBanner/DismissableBanner.tsx

```diff
@@ -102,23 +102,34 @@ export const DismissableBanner = (props: Props) => {
   const classes = useStyles();
   const storageApi = useApi(storageApiRef);
   const notificationsStore = storageApi.forBucket('notifications');
-  const rawDismissedBanners =
-    notificationsStore.snapshot<string[]>('dismissedBanners').value ?? [];
+  const observedItems = useObservable(
+    notificationsStore.observe$<string[]>('dismissedBanners'),
+    notificationsStore.snapshot<string[]>('dismissedBanners'),
+  );
 
   const [dismissedBanners, setDismissedBanners] = useState(
-    new Set(rawDismissedBanners),
+    new Set(observedItems.value ?? []),
   );
-
-  const observedItems = useObservable(
-    notificationsStore.observe$<string[]>('dismissedBanners'),
+  const [loadingSettings, setLoadingSettings] = useState(
+    observedItems.presence === 'unknown',
   );
```

Comment:
One more thing - this doesn't need a useState at all right? It's only a computed property of `observedItems`. So you could probably remove basically all code below and instead just do like `const loadingSettings = observedItems.presence === 'unknown';` and then use that directly where needed.

Reply:
It's the first approach I've tried, but it still resulted in some layout shift.

I'll see if I can improve this.